### PR TITLE
GetUnconfirmedActionsByAddressRequest don't include the unconfirmed a…

### PR DESCRIPTION
**Pull request by lzxm160**
…ctions to the given address #1231

### CODE REVIEW

File: actpool/actpool.go [hunk 1]
```diff
@@ -240,11 +241,22 @@ func (ap *actPool) GetPendingNonce(addr string) (uint64, error) {
...
+	var ret []action.SealedEnvelope
 	if queue, ok := ap.accountActs[addr]; ok {
-		return queue.AllActs()
+		ret = queue.AllActs()
+	}
+	for _, action := range ap.allActions {
+		dst, ok := action.Destination()
+		pubKeyHash := action.SrcPubkey().Hash()
+		srcAddr, _ := address.FromBytes(pubKeyHash)
+		if ok && strings.EqualFold(dst, addr) && !strings.EqualFold(srcAddr.String(), addr) {
```
Comment (zjshen14) [inline actpool/actpool.go]: let's use address.Equal

File: actpool/actpool.go [hunk 2]
```diff
@@ -240,11 +241,22 @@ func (ap *actPool) GetPendingNonce(addr string) (uint64, error) {
...
+			ret = append(ret, action)
+		}
+	}
+	if len(ret) == 0 {
```
Comment (zjshen14) [inline actpool/actpool.go]: Doesn't matter

File: actpool/actpool.go (hunk 1 above)
Comment (zjshen14) [inline actpool/actpool.go]: It seems that you can check the 3 conditions one by one, and continue earlier

File: actpool/actpool_test.go [hunk 3]
```diff
@@ -959,19 +959,22 @@ func TestActPool_GetUnconfirmedActs(t *testing.T) {
...
 	require.NoError(err)
 	err = ap.Add(tsf4)
 	require.NoError(err)
+	err = ap.Add(tsf5)
+	require.NoError(err)
 
-	acts := ap.GetUnconfirmedActs(addr2)
-	require.Equal([]action.SealedEnvelope{}, acts)
+	acts := ap.GetUnconfirmedActs(addr3)
+	require.Equal([]action.SealedEnvelope(nil), acts)
```
Comment (zjshen14) [inline actpool/actpool_test.go]: Why nil?

[]action.SealedEnvelope{} doesn't work?

File: actpool/actpool.go [hunk 4]
```diff
@@ -240,11 +240,33 @@ func (ap *actPool) GetPendingNonce(addr string) (uint64, error) {
...
+	var ret []action.SealedEnvelope
 	if queue, ok := ap.accountActs[addr]; ok {
-		return queue.AllActs()
+		ret = queue.AllActs()
+	}
+	addressAction, err := address.FromString(addr)
+	if err != nil {
+		return nil
+	}
+	for _, action := range ap.allActions {
```
Comment (zjshen14) [inline actpool/actpool.go]: Can you leave a todo comment and link to the github issue for the performance concern?

**Comment by zjshen14** [comment]
This is an incompatible change in terms of API semantics. cc @puncsky

File: actpool/actpool_test.go (hunk 3 above)
Comment (lzxm160) [reply actpool/actpool_test.go]: []action.SealedEnvelope{} doesn't work,I think nil is different from empty

File: actpool/actpool.go (hunk 4 above)
Comment (lzxm160) [reply actpool/actpool.go]: sure

**Pull request merged**